dallinger/models.py

```diff
@@ -68,6 +68,21 @@ class SharedMixin(object):
     #: a generic column for storing structured JSON data
     details = Column(JSONB, nullable=False, server_default="{}", default=lambda: {})
 
+    def __json__(self):
+        """Return json description of a participant."""
+        return {
+            "id": self.id,
+            "creation_time": self.creation_time,
+            "failed": self.failed,
+            "time_of_death": self.time_of_death,
+            "property1": self.property1,
+            "property2": self.property2,
+            "property3": self.property3,
+            "property4": self.property4,
+            "property5": self.property5,
+            "details": self.details,
+        }
+
```

Comment:
I don't feel strongly about it, but I wonder if the base class's `__json__()` method should call a pre-defined hook on the subclasses to extend the basic set:
```
def __json__(self):
    base = {
        "id": self.id,
        "creation_time": self.creation_time,
        "failed": self.failed,
        "time_of_death": self.time_of_death,
        "property1": self.property1,
        "property2": self.property2,
        "property3": self.property3,
        "property4": self.property4,
        "property5": self.property5,
        "details": self.details,
    }
    return base.extend(self._json_extra())
```

It would make it more obvious what each model adds that's unique to it.

Reply:
I definitely think that's the way to go and considered doing something similar, but I wanted to be minimally invasive. Probably best to do it in the way that is most likely to prevent this kind of thin from coming up again though.

Reply:
This should be addressed to your satisfaction.